### PR TITLE
65 dev update to latest squirelib

### PR DESCRIPTION
**Pull request by akbulutdora**
Fixes the errors on the main branch with `SquireLib`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -29,6 +30,7 @@ uuid = { version = "1.1", features = ["serde", "v4"] }
 reqwest = { version = "0.11.10", features = ["json"] }
 dotenv = { version = "0.9" }
 tempfile = { version = "3.3.0" }
+num-rational = { version = "0.4.1", features = ["serde"] }
```
Comment (TylerBloom) [inline Cargo.toml:33]: Do you need this? `squire_lib` should be re-exporting any necessary `r64`.

File: src/model/guilds/guild_tournament.rs [hunk 2]
```diff
@@ -872,18 +844,18 @@ impl GuildTournament {
                 return Ok(digest);
             }
         };
-        let op = TournOp::PlayerOp(p_id, RecordResult(result.clone()));
+        let r_id = self
+            .tourn
+            .round_reg
+            .get_player_active_round(&p_id)
+            .unwrap()
```
Comment (TylerBloom) [inline src/model/guilds/guild_tournament.rs:851]: Should we unwrap here? What if the player does not have an active round?

File: src/model/guilds/guild_tournament.rs [hunk 3]
```diff
@@ -918,18 +890,18 @@ impl GuildTournament {
                 return Ok(digest);
             }
         };
-        let op = TournOp::PlayerOp(p_id, ConfirmResult);
+        let r_id = self
+            .tourn
+            .round_reg
+            .get_player_active_round(&p_id)
+            .unwrap()
```
Comment (TylerBloom) [inline src/model/guilds/guild_tournament.rs:897]: Ditto

File: src/setup_commands/defaults_commands.rs [hunk 4]
```diff
@@ -659,6 +659,10 @@ async fn standard(ctx: &Context, msg: &Message, _: Args) -> CommandResult {
     subcommand_default(ctx, msg, "settings defaults tournament scoring standard").await
 }
 
+#[allow(non_camel_case_types)]
+/// The numerical type used in the scoring systems
+pub type r64 = num_rational::Rational32;
```
Comment (TylerBloom) [inline src/setup_commands/defaults_commands.rs:664]: You should be able to use the `r64` defined here: `squire_lib::r64`.

File: src/setup_commands/defaults_commands.rs [hunk 5]
```diff
@@ -670,7 +674,7 @@ async fn standard(ctx: &Context, msg: &Message, _: Args) -> CommandResult {
 async fn match_win_points(ctx: &Context, msg: &Message, mut args: Args) -> CommandResult {
     use squire_lib::settings::StandardScoringSetting::*;
     let data = ctx.data.read().await;
-    match args.single_quoted::<f64>() {
+    match args.single_quoted::<r64>() {
```
Comment (TylerBloom) [inline src/setup_commands/defaults_commands.rs:677]: It should be the case that this correctly parses strings like `1`, `0.5`, and `-2`, but I'm not sure. Did you test this?

File: Cargo.toml (hunk 1 above)
Comment (akbulutdora) [reply Cargo.toml:33]: You're right. I will import from `squire_lib`.

File: src/model/guilds/guild_tournament.rs (hunk 2 above)
Comment (akbulutdora) [reply src/model/guilds/guild_tournament.rs:851]: I didn't write this myself, I merely moved it around.  unwrap was used in the `Ok()` arm of the following match expression. I don't know if fixing this sits with the general theme of updating to the latest `squire_lib`. What do you think?

```
match self.tourn.apply_op(Utc::now(), op)
```

File: src/setup_commands/defaults_commands.rs (hunk 5 above)
Comment (akbulutdora) [reply src/setup_commands/defaults_commands.rs:677]: I followed `squire_lib` instead of testing myself. I will test it.

**Review by TylerBloom — APPROVED** [review]
Looks good to me. If these are all of the required changes to update to the lastest `squire_lib`, please squash and rebase.

**Pull request merged**